### PR TITLE
Improve the way we deliver Examples in command line.

### LINKED ISSUE #15869: Improve the way we deliver Examples in command line
The build and run commands both have long sections of examples.  In run them come in the middle of the page in build they come at the end.

- Standardize examples at the end of every page
- Design a consistent structure for calling out "categories of examples" in these larger sections

http://docs.docker.com/reference/commandline/build/
http://docs.docker.com/reference/commandline/run/

### PR DESCRIPTION
**Pull request by zhangjianfnst**
Categorize the larger section examples by topic of [OPTION] usage . Fixes #15869

Signed-off-by: Jian Zhang zhangjian.fnst@cn.fujitsu.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Oh, terribly sorry for us being so late reviewing. Looks like this needs a rebase, @zhangjianfnst :cry:

**Comment by zhangjianfnst** [comment]
@thaJeztah I have done a rebase, please check it. Thank you!

**Comment by SvenDowideit** [comment]
LGTM - @moxiegirl @thaJeztah

File: docs/reference/commandline/build.md [hunk 1]
```diff
@@ -263,7 +281,7 @@ depending on which host they build an image on.
 
 A good example is `http_proxy` or source versions for pulling intermediate
 files. The `ARG` instruction lets Dockerfile authors define values that users
-can set at build-time using the  `---build-arg` flag:
+can set at build-time using the  `--build-arg` flag:
```
Comment (thaJeztah) [inline docs/reference/commandline/build.md]: :+1:

**Comment by thaJeztah** [comment]
I like the improved structure, but still on the fence if we should use the flag-name for the headings, or try to create a descriptive title (e.g. What we had on https://github.com/docker/docker/pull/16617/files#diff-505c72218d90da970c16fdbf0b4f613cL538)

**Comment by moxiegirl** [comment]
@zhangjianfnst Nicely done.  A big step forward.  I agree with @thaJeztah thought that descriptive titles would be more help.  You could combine both, for example:
```
run
Examples
  Allocate psuedo-TTY (–name, -it)
  Capture container id (–cidfile)
  Full container capabilities (–privileged)
  Set commmand path (-w)
```

**Comment by thaJeztah** [comment]
Thanks for the suggestion @moxiegirl, I like that :+1:

**Comment by zhangjianfnst** [comment]
@moxiegirl @thaJeztah Thanks for your suggestion, I will update my PR.

**Comment by zhangjianfnst** [comment]
@moxiegirl @thaJeztah I added descriptive titles, please check it.

File: docs/reference/commandline/run.md [hunk 2]
```diff
@@ -132,11 +138,15 @@ lifts all the limitations enforced by the `device` cgroup controller. In other
...
 flag exists to allow special use-cases, like running Docker within Docker.
 
+### Set working directory (-w)
+
     $ docker  run -w /path/to/dir/ -i -t  ubuntu pwd
 
 The `-w` lets the command being executed inside directory given, here
 `/path/to/dir/`. If the path does not exists it is created inside the container.
 
+### Mount volumn (-v, --read-only)
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md]: typo: s/volumn/volume/

File: docs/reference/commandline/run.md [hunk 3]
```diff
@@ -179,6 +191,7 @@ publishing the port to the host system's interfaces. The [Docker User
 Guide](../../userguide/dockerlinks.md) explains in detail how to manipulate
 ports in Docker.
 
+### Set environment variables (-e, --env, --env-file)
     $ docker run -e MYVAR1 --env MYVAR2=foo --env-file ./env.list ubuntu bash
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md]: can you add a blank like here?

**Comment by moxiegirl** [comment]
@zhangjianfnst thank you for that.   This looks good. LGTM

**Comment by thaJeztah** [comment]
Thanks so much @zhangjianfnst, great job and a great suggestion for improvement!

LGTM

**Comment by zhangjianfnst** [comment]
@moxiegirl @thaJeztah That's great, my first PR is merged! :smiley:

**Comment by thaJeztah** [comment]
congrats @zhangjianfnst, and thank you for contributing!

:dancers: :tada:

**Comment by moxiegirl** [comment]
cherrypick

**Pull request merged**